pkg/serializer/serializer.go

```diff
@@ -312,8 +312,10 @@ func (s *Serializer) SendIterableSeries(series *metrics.IterableSeries) error {
 	var extraHeaders http.Header
 	var err error
 
-	if useV1API {
+	if useV1API && s.enableJSONStream {
 		seriesPayloads, extraHeaders, err = s.serializeIterableStreamablePayload(seriesSerializer, stream.DropItemOnErrItemTooBig)
+	} else if useV1API && !s.enableJSONStream {
```

Comment:
Is there any reason to support this in the new code? Is there a benefit to use this variant of json marshaller over the streaming one?

Reply:
Unfortunately, `serializeIterableStreamablePayload` requires `zlib` tag. See https://github.com/DataDog/datadog-agent/blob/ff6ab4a990c17285bf9ad2e31766f57ab0a6f9c6/pkg/serializer/serializer.go#L137 and https://github.com/DataDog/datadog-agent/blob/ff6ab4a990c17285bf9ad2e31766f57ab0a6f9c6/pkg/serializer/stream/compressor.go#L6-L25

`zlib` tag is not set for several flavors of the Agent and it is not trivial to set it for the serverless agent for example.